libs/core/threading_base/include/hpx/threading_base/annotated_function.hpp

```diff
@@ -214,7 +230,7 @@ namespace hpx { namespace util {
     ///
     /// \param function
     template <typename F>
-    F&& annotated_function(F&& f, char const* = nullptr)
+    F&& annotated_function(F&& f, std::string = "")
```

Comment:
I think we should leave the overload that takes a `char const*` as otherwise the description will always cause an allocation (even if the string is discarded right away).

Reply:
Fair point! I'll change that.